### PR TITLE
Cd/classes

### PR DESCRIPTION
**Pull request by Carmen-Dominguez**
- switched to using express instead
- added the following classes
  - location
  - user
  - schedule
- created the following controllers (wip)
  - schedule
  - location
- created post and get for location

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,13 +22,15 @@
   "license": "ISC",
   "dependencies": {
     "dotenv": "^16.0.0",
+    "express": "^4.21.1",
```
Comment (slampunk) [inline package.json]: weird, i thought it'd pull down v5

File: src/controllers/location.controller.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (slampunk) [inline src/controllers/location.controller.ts:1]: You can remove this line since you're using `Location` later on

File: package.json (hunk 1 above)
Comment (Carmen-Dominguez) [reply package.json]: Looks like Express 5.0 requires Node.js 18 or higher, which still doesn't make sense cuz I got v18.18.1

File: src/main.ts [hunk 3]
```diff
@@ -1,38 +1,47 @@
...
+
+  // input is correct type
+  if (locationController.isLocation(location)) {
+    // location is new
+    const l: Location = { name: location.name.toLowerCase().trim(), seats: location.seats }
+
+    if (locationController.getLocation(l.name) === undefined) {
+      locationController.createLocation(l)
+      res.status(201).json({ name: location.name, seats: location.seats })
+    } else res.status(418).json({ error: `${l.name} already exists` })
```
Comment (slampunk) [inline src/main.ts]: you like teapots hey 😂

scope out the 409 response on wikipedia 😉

File: src/main.ts [hunk 4]
```diff
@@ -1,38 +1,47 @@
...
+  } else res.status(422).json({ error: 'wrong data type' })
+})
+
+app.get('/locations', (req, res) => {
+  const locations = locationController.getAllLocations()
+  res.status(201).json(locations)
```
Comment (slampunk) [inline src/main.ts]: you can make this a `200`. `201` is used when creating resources, but here you're simply _getting_ (i.e. not creating) a resource

**Review by slampunk — COMMENTED** [review]
couple of minor comments and tips

File: package.json (hunk 1 above)
Comment (slampunk) [reply package.json]: not a biggie. we can circle back later on if you wana pull off node v20

**Review by slampunk — APPROVED** [review]
LGTM

**Pull request merged**